### DISCUSSION TITLE
Add optional flags to nova:user

### DISCUSSION THREAD

**Discussion by garritfra** [laravel/nova-issues#5511] [Ideas & Feature Requests] [answered]
## Problem

I just ran into an issue with running Laravel in a serverless environment, where it's impossible to run interactive commands, like `nova:user`. Using the `--no-interactive` flag lets the command run through, but I can't configure the username, email and password.

## Use Case

I want to be able to run the `nova:user` command with username, email and password flags as opposed to interactive prompts.

## Current Workaround

As a workaround, I patched this feature into my local `nova:user` command in the `vendor` directory and deployed the modified code to AWS lambda, and invoked the function with the proper flags.

**Comment by crynobone** [comment]
This is something that we had discuss internally and feel that isn't something that we can provide out of the box:

1. You maybe able to customize the command to get additional input and set the value in the database.
2. Setting `--password value` flag via vapor console etc will make the value available in Cloudwatch and may pose a security concern.

However, you may look into project such as https://github.com/nova-kit/nova-on-vapor#new-novavapor-user-command which offer similar features

**Reply by garritfra** [reply]
I'm aware that setting the password via a flag is insecure and immediately changed it through the UI once the user was created.

Maybe that could be addressed by adding an explicit warning about this insecure behaviour, and let the user decide if it's a good idea or not.

Creating a custom command just to bootstrap a user seems rather overkill. If the framework already provides a command for this, shouldn't it cover all use cases?

**Reply by crynobone** [reply, marked as answer]
Feel free to create your own command, it's not an overkill as you imagine 😉
